src/NBitcoin/Block.cs

```diff
@@ -241,23 +258,36 @@ public Block()
             this.SetNull();
         }
 
+        public Block(NetworkOptions networkOptions)
+            :this(new BlockHeader(networkOptions))
+        {            
+        }
+
         public Block(BlockHeader blockHeader)
         {
-            this.SetNull();
             this.header = blockHeader;
         }
 
-        public Block(byte[] bytes)
+        public Block(byte[] bytes, NetworkOptions networkOptions = null)
+            :this(networkOptions)
         {
-            this.ReadWrite(bytes);
+            this.ReadWrite(bytes, options:this.GetNetworkOptions());
         }
 
+        public NetworkOptions GetNetworkOptions()
+        {
+            return this.header.GetNetworkOptions();
+        }
 
         public void ReadWrite(BitcoinStream stream)
         {
+            if (stream.Serializing)
+                stream.TransactionOptions = this.GetNetworkOptions();
+
             stream.ReadWrite(ref this.header);
             stream.ReadWrite(ref this.transactions);
-            if (Block.BlockSignature)
+
+            if (stream.TransactionOptions.IsProofOfStake)
```

Comment:
You could achieve this without passing Options in the constructor right?

Reply:
That would always be possible - the question is what are we trading for that. For one it does mean that any methods that do need the options would have to have those options supplied. It seems preferable to just pass it once via the constructor. That would enable us to do things such as Block.CreateTransaction or a parameterless Block.AddTransaction - all based on the same consistent options specified once. If we don't add it to the constructor, then whenever we pass a block to another method we would need to also pass along the options that we want associated with the block.